src/main/java/com/devcycle/sdk/server/openfeature/DevCycleProvider.java

```diff
@@ -23,6 +28,28 @@ public Metadata getMetadata() {
         return () -> PROVIDER_NAME + " " + devcycleClient.getSDKPlatform();
     }
 
+    @Override
+    public void initialize(EvaluationContext evaluationContext) throws Exception {
+        if (devcycleClient.isInitialized()) {
+            return;
+        }
+
+        long deadline = 2 * 1000; // Delay in milliseconds
+        long start = System.currentTimeMillis();
+
+        do {
+            if (deadline <= System.currentTimeMillis() - start) {
+                throw new GeneralError("DevCycle client not initialized within 2 seconds");
+            }
+            Thread.sleep(5);
+        } while (!devcycleClient.isInitialized());
```

Comment:
this is how it's done in Flagd provider and other places...

Reply:
don't we have a way of doing our initialization callback here? That would be cleaner

Reply:
well, no, this function is also synchronous and wouldn't do anything with a callback here. https://openfeature.dev/docs/reference/technologies/server/java#synchronous